implementations/rust/ockam/ockam_command/src/credential/store.rs

```diff
@@ -90,25 +115,29 @@ async fn run_impl(
         .terminal
         .progress_output(&output_messages, &is_finished);
 
-    let (cred_as_str, _) = try_join!(send_req, progress_output)?;
-
-    opts.terminal
-        .stdout()
-        .machine(cred_as_str.to_string())
-        .json(serde_json::json!(
-            {
-                "name": cmd.credential_name,
-                "issuer": cmd.issuer,
-                "credential": cred_as_str
-            }
-        ))
-        .plain(fmt_ok!(
-            "Credential {} stored\n",
-            cmd.credential_name
-                .to_string()
-                .color(OckamColor::PrimaryResource.color())
-        ))
-        .write_line()?;
-
-    Ok(())
+    let ((is_valid, result), _) = try_join!(send_req, progress_output)?;
+
+    if !is_valid {
+        Err(miette!(result).into())
+    } else {
```

Comment:
If you apply the suggestion from the other comment, then this part also gets simplified:

```rust
let (credential, _) = try_join!(send_req, progress_output)?;
opts.terminal
    .stdout()
...
```